contracts/Collector.sol

```diff
@@ -3,17 +3,17 @@ pragma solidity ^0.6.12;
 pragma experimental ABIEncoderV2;
 
 import "./interfaces/ICollector.sol";
-import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20Burnable.sol";
 import "@openzeppelin/contracts/math/SafeMath.sol";
 
 contract Collector is ICollector{
-    IERC20 public token;
+    ERC20Burnable public token;
```

Comment:
Although it makes sense, I'm not sure we can use this interface. It extends the [EIP-20 standard](https://eips.ethereum.org/EIPS/eip-20) with additional functions that may be not implemented by the token.

Reply:
This is the key part. If not all tokens implement the burnable method then this proposal falls apart and I will think about something different.